File: Array of Products/O(n)_time_and_space.py
def arrayOfProducts(array):
    # Write your code here.
    pre=1
    post=1
    arr=[1]*len(array)
    for i in range(len(array)):
        arr[i] *= pre
        pre *= array[i]
        arr[len(array)-1-i] *= post
        post *= array[len(array)-1-i]
    return arr

File: Arrays/Array of Products/O(n)_time_and_space.js
function arrayOfProducts(array) {
  // Write your code here.
  let post = 1;
  let pre = 1;
  let ret = new Array(array.length).fill(1);

  for (let i = 0; i < array.length; i++) {
    ret[i] = ret[i] * pre;
    pre = pre * array[i];
    ret[array.length - i - 1] = ret[array.length - i - 1] * post;
    post = post * array[array.length - i - 1];
  }
  return ret;
}
